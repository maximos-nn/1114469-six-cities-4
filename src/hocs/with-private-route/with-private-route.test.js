import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import history from "../../history";
import withPrivateRoute from "./with-private-route";

const mockStore = configureStore([]);

const MockComponent = () => <div />;
const MockComponentWrapped = withPrivateRoute(MockComponent, `AUTH`, `/login`);

it(`Wrapped component renders correctly`, () => {
  const store = mockStore({USER: {authenticationStatus: `AUTH`}});
  const tree = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <MockComponentWrapped />
        </Router>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
