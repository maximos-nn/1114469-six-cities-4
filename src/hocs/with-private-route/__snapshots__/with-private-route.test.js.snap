// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapped component renders correctly 1`] = `<div />`;
